compactor/compactor.go

```diff
@@ -124,10 +128,13 @@ func (t *Periodic) Resume() {
 	t.paused = false
 }
 
-func (t *Periodic) getRev(h int) int64 {
+func (t *Periodic) getRev(h int) (int64, []int64) {
 	i := len(t.revs) - int(time.Duration(h)*time.Hour/checkCompactionInterval)
 	if i < 0 {
-		return -1
+		return -1, t.revs
+	}
+	if i >= len(t.revs) {
```

Comment:
With the current code, it's evaluated only when `h` is `0`.
It was for just in case, so I removed it.